underlay/src/main/java/bio/terra/tanagra/serialization/UFUnderlay.java

```diff
@@ -24,6 +25,7 @@ public class UFUnderlay {
   private final String primaryEntity;
   private final String uiConfig;
   private final String uiConfigFile;
+  private final Map<String, UFPluginConfig> plugins;
```

Comment:
I tried doing this originally, but it creates a circular dependency between `tanagra.service` and `tanagra.underlay`. Are you suggesting moving `PluginType` to `tanagra.underlay` and referencing it from there?

Reply:
That also creates a circular dependency.

Reply:
`PluginType` references the base `Plugin` interface, the interfaces for the individual plugins, as well as the default implementations for every plugin. If I move `PluginType` to `plugin.underlay` it will need to import those, which creates the circular dependency. We could probably work around the interface issue, but the default implementations are always going to be a problem assuming you think they belong in core.

Reply:
Ah okay I understand the problem now, thanks. I guess we could just put the enum in that package and skip the class mapping. My intention around the class mapping was to avoid the caller having to know the class name, and instead letting you just pull it off the enum, but since you're not doing that here anyway, I don't see much point in that. For this PR, fine with me to do either: remove the class name from the enum since you're not using anyway and move `PluginType` to `underlay` package, or just leave it as is (`string -> string` map in the config schema) with a `TODO` comment to enforce this later.